### PR TITLE
Fix GCC -Wextra warnings

### PR DESCRIPTION
**Pull request by ozone10**
fix conversion warnings
fix missing-field-initializers warnings
fix unused-parameter warnings
fix deprecated-copy warnings
add initializers

fix https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13122

### CODE REVIEW

File: PowerEditor/src/MISC/Common/Common.cpp [hunk 1]
```diff
@@ -1690,7 +1690,7 @@ void Version::setVersionFrom(const generic_string& filePath)
 			return;
 
 		unsigned char* buffer = new unsigned char[bufferSize];
-		::GetFileVersionInfo(filePath.c_str(), uselessArg, bufferSize, buffer);
```
Comment (donho) [inline PowerEditor/src/MISC/Common/Common.cpp:1693]: I don't think it makes warning. Does it?

File: PowerEditor/src/WinControls/FunctionList/functionParser.cpp [hunk 2]
```diff
@@ -564,7 +564,7 @@ size_t FunctionZoneParser::getBodyClosePos(size_t begin, const TCHAR *bodyOpenSy
 	return targetEnd;
 }
 
-void FunctionZoneParser::classParse(vector<foundInfo> & foundInfos, vector< pair<size_t, size_t> > &scannedZones, const std::vector< std::pair<size_t, size_t> > & commentZones, size_t begin, size_t end, ScintillaEditView **ppEditView, generic_string classStructName)
+void FunctionZoneParser::classParse(vector<foundInfo> & foundInfos, vector< pair<size_t, size_t> > &scannedZones, const std::vector< std::pair<size_t, size_t> > & commentZones, size_t begin, size_t end, ScintillaEditView **ppEditView, generic_string /*classStructName*/)
```
Comment (donho) [inline PowerEditor/src/WinControls/FunctionList/functionParser.cpp:567]: Could you also rename `classStructName` to `subLevelClassStructName` in line 585 & 606 please?

**Review by donho — CHANGES_REQUESTED** [review]
Could you also rename classStructName to subLevelClassStructName in line 585 & 606 please?

File: PowerEditor/src/WinControls/FunctionList/functionParser.cpp (hunk 2 above)
Comment (ozone10) [reply PowerEditor/src/WinControls/FunctionList/functionParser.cpp:567]: done

Comment (donho) [reply PowerEditor/src/WinControls/FunctionList/functionParser.cpp:567]: Thank you!

**Review by donho — APPROVED** [review]

**Pull request closed without merging**